EIPS/eip-overlay_tree.md

```diff
@@ -0,0 +1,108 @@
+
+## Abstract
+
+This EIP describes a four phase process to complete the conversion.
+
+  * In the first phase, all new state writes are made to an overlay binary trie, while the hexary trie is being converted to binary. The block format is changed to have two storage roots: the root of the hexary trie and the root of the binary trie.
+  * The second phase starts when miners are done with the conversion and replace the hexary root with the newly calculated root.
+  * The third phase begins when a sufficient number of consecutive blocks report the same value for `binary_trie_root`: the overlay tree is progressively merged back into the newly converted binary base trie. A constant number of entries are deleted from the overlay and inserted into the base trie.
+  * The fourth and final phase begins when the overlay trie is empty. The field holding its root is removed from the block header.
+
+## Motivation
+
+There is a long running interest in switching the state trie from a hexary format to a binary format, for reasons pertaining to proof and storage sizes. The conversion process poses a catch-up issue, caused by the sheer size of the full state: it can not be translated in a reasonable time (i.e. on the same order of magnitude as the block time). 
+
+## Specification
+The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).
+
+### Binary tries
+
+This EIP assumes that a binary trie is defined like the MPT, except that:
+
+  * The series of bytes in I₀ is seen as a series of _bits_ and so ∀i≤256, I₀[i] is the ith bit in key I₀
+  * The first item of an **extension** or a **leaf** is replacing nibbles with bits;
+  * A **branch** is a 2 item structure in which both items correspond to each of the two possible bit values for the keys at this point in their traversal;
+  * c(𝕴,i) ≡ RLP((u(0), u(1)) at a branch, where u(j) = n({I : I ∈ 𝕴 ⋀ I₀[i] = j}, i+1)
+
+### Phase 1
+
+Let _H₁_ be the previously agreed-upon block height at which phase 1 starts. For each block of height H₁ ≤ _h_ < H₂:
+
+  * Block headers contain a new Hₒ field, which is the _root of the overlay binary trie_;
+  * Hᵣ ≡ P(H)ᵣ, i.e. the hexary trie root is the same as that of the block parent's. The hexary trie is referred to as the _hexary base trie_.
+
+The following is changed in the execution environment:
+
+  * Upon executing a _state read_, ϒ first searches for the address in the overlay trie. If the key can not be found there, ϒ then searches the hexary base trie as it did at block heights h' < H₁;
+  * Upon executing a _state write_, ϒ will insert or update the value into the overlay tree.
```

Comment:
Shouldn't this also clarify that:
1) Before every write a read also happens (needed to determine costs).
2) Upon writing into the overlay, it is deleted from the base trie? 

I'm not sure if 2) is proposed here or whether it is a good idea or not, but clarity would be nice.

Reply:
During phase 1, values are not deleted from the base trie, as the base trie is read-only. I'll make it clear :+1: